raptor/src/main/java/org/opentripplanner/raptor/relaxedlimitedtransfer/RelaxedLimitedTransferSearch.java

```diff
@@ -0,0 +1,234 @@
+/// in the main raptor search.
+public class RelaxedLimitedTransferSearch<T extends RaptorTripSchedule> {
+
+  private final RaptorTransitDataProvider<T> data;
+  private final RaptorTransitCalculator<T> transitCalculator;
+  private final int earliestDepartureTime;
+  private final int latestDepartureTime;
+  private final Collection<RaptorAccessEgress> accesses;
+  private final Collection<RaptorAccessEgress> egresses;
+  private final RelaxFunction relaxFunction;
+
+  /* Variables used during the search (mutable) */
+
+  private int currentRouteBoardSlack = 0;
+
+  public RelaxedLimitedTransferSearch(
+    Collection<RaptorAccessEgress> accesses,
+    Collection<RaptorAccessEgress> egresses,
+    RaptorTransitDataProvider<T> data,
+    int earliestDepartureTime,
+    int searchWindowInSeconds,
+    RaptorTransitCalculator<T> transitCalculator,
+    RelaxFunction costRelaxFunction
+  ) {
+    this.data = data;
+    this.transitCalculator = transitCalculator;
+    this.earliestDepartureTime = earliestDepartureTime;
+    this.latestDepartureTime = earliestDepartureTime + searchWindowInSeconds;
+    this.accesses = accesses;
+    this.egresses = egresses;
+    this.relaxFunction = costRelaxFunction;
+  }
+
+  /// Run the search
+  public Collection<RaptorPath<T>> route() {
+    var results = new ParetoSet<RaptorPath<T>>(new DestinationArrivalComparator<>(relaxFunction));
+
+    var routes = data.routeIndexIterator(findAllAccessStopIndexes());
+
+    while (routes.hasNext()) {
+      var route = data.getRouteForIndex(routes.next());
+      var paths = routeSearch(route);
+      results.addAll(paths);
+    }
+    return results;
+  }
+
+  private IntIterator findAllAccessStopIndexes() {
+    BitSet accessStopBitSet = new BitSet();
+    for (RaptorAccessEgress it : accesses) {
+      accessStopBitSet.set(it.stop());
+    }
+    return new BitSetIterator(accessStopBitSet);
+  }
+
+  /// First find ONE path for each combination of access/egress. This corresponds to the
+  /// first iteration of a RangeRaptor search. We will later expand this to all trip schedules
+  /// within the search-window. All paths with the same (route, access, and egress) will have
+  /// almost identical cost, so we can use this to prune the set of paths before expanding the
+  /// timetable. For each route/pattern we only want the best combination of access and egress.
+  private List<RaptorPath<T>> routeSearch(RaptorRoute<T> route) {
+    this.currentRouteBoardSlack = data.slackProvider().boardSlack(route.pattern().slackIndex());
+    RaptorPath<T> bestPath = null;
+
+    for (var access : accesses) {
+      var pattern = route.pattern();
+      int boardPos = pattern.findStopPositionAfter(0, access.stop());
+
+      if (boardPos == -1) {
+        continue;
+      }
+
+      for (var egress : egresses) {
+        int alightPos = pattern.findStopPositionAfter(boardPos + 1, egress.stop());
+
+        if (alightPos == -1) {
```

Comment:
There is an old technical depth thing here. The RaptorPattern should return `RaptorConstants.NOT_FOUND` here and not `-1`. Since the constant is not `-1` this is a small refactoring. I do not expect to fix it in this PR, but it be nice to get it fixed at some point.